src/mito2/src/memtable/merge_tree/data.rs

```diff
@@ -186,6 +335,8 @@ impl DataBuffer {
                 })
                 .push_value_ref(field);
         }
+
+        self.ts_builder.len() >= self.freeze_threshold
```

Comment:
Maybe we should provide a `len()` method and check this outside of the buffer. Then we don't need to store the threshold in the buffer.